actor/builtin/paymentbroker/paymentbroker.go

```diff
@@ -750,18 +760,29 @@ func findByChannelLookup(ctx context.Context, storage exec.Storage, byPayer exec
 
 // checkCondition combines params in the condition with the redeemerSuppliedParams, sends a message
 // to the actor and method specified in the condition, and returns an error if one exists.
-func checkCondition(vmctx exec.VMContext, condition *types.Predicate, redeemerSuppliedParams []interface{}) (uint8, error) {
+func checkCondition(vmctx exec.VMContext, channel *PaymentChannel, condition *types.Predicate, redeemerSuppliedParams []interface{}) error {
 	if condition == nil {
-		return 0, nil
+		if channel != nil {
+			channel.Condition = nil
+		}
+		return nil
+	}
+
+	if !channel.Redeemed || channel.Condition == nil || len(redeemerSuppliedParams) > 0 {
```

Comment:
BLOCKING: This will panic if channel doesn't exist (wrong channel id). `checkCondition` should only be called after we've established the existence of the channel in `updateChannel`. Also these conditions need a comment to explain the logic. e.g.:

// Update payment channel condition if this is the first redeem, we haven't had a condition previously, or the
// redeemer has provided new parameters.

Reply:
It should no longer be possible to call `checkCondition` without a channel. The prior nil check was a mistaken leftover from some previous code.

I'll add a comment.